Composer/packages/lib/shared/__tests__/copyUtils/copyAdaptiveAction.test.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { copyAdaptiveAction } from '../../src/copyUtils';
+import { CopyConstructorMap } from '../../src/copyUtils/copyAdaptiveAction';
+import { externalApiStub as externalApi } from '../jestMocks/externalApiStub';
+import { SDKTypes } from '../../src';
+
+describe('copyAdaptiveAction', () => {
+  it('should return {} when input is invalid', async () => {
+    expect(await copyAdaptiveAction('hello', externalApi)).toEqual('hello');
+
+    expect(await copyAdaptiveAction(null as any, externalApi)).toEqual({});
+    expect(await copyAdaptiveAction({} as any, externalApi)).toEqual({});
+    expect(await copyAdaptiveAction({ name: 'hi' } as any, externalApi)).toEqual({});
+  });
+
+  describe('should call certain handler in CopyConstructorMap', () => {
+    let originCopyConstructorMap = {};
+    beforeAll(() => {
+      originCopyConstructorMap = { ...CopyConstructorMap };
+    });
+
+    beforeEach(() => {
+      Object.keys(CopyConstructorMap).forEach(key => {
+        CopyConstructorMap[key] = jest.fn();
+      });
+    });
```

Comment:
This is better done with jest module mocking.

```ts
jest.mock('../../src/copyUtils/copyAdaptiveAction', () => ({
  'Microsoft.TextInput': jest.fn(),
  ...otherTypes
});

// in a test
expect(CopyConstructorMap['Microsoft.TextInput']).toHaveBeenCalledTimes(1);
```

The path may need to be the exact same one used in the app code. I forget.